Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -101,42 +101,51 @@ theorem card_fintypeIic : Fintype.card (Set.Iic b) = b + 1 := by
 
 theorem card_fintypeIio : Fintype.card (Set.Iio b) = b := by rw [Fintype.card_ofFinset, card_Iio]
 
--- TODO@Yaël: Generalize all the following lemmas to `SuccOrder`
+@[deprecated "Use `Finset.Icc_succ_left_eq_Ioc`" (since := "2025-02-25")]
 theorem Icc_succ_left : Icc a.succ b = Ioc a b := by
   ext x
   rw [mem_Icc, mem_Ioc, succ_le_iff]
 
+@[deprecated "Use `Finset.Ico_succ_right_eq_Icc`" (since := "2025-02-25")]
 theorem Ico_succ_right : Ico a b.succ = Icc a b := by
   ext x
   rw [mem_Ico, mem_Icc, Nat.lt_succ_iff]
 
+@[deprecated "Use `Finset.Ico_succ_left_eq_Ioo`" (since := "2025-02-25")]
 theorem Ico_succ_left : Ico a.succ b = Ioo a b := by
   ext x
   rw [mem_Ico, mem_Ioo, succ_le_iff]
 
+@[deprecated "Use `Finset.Icc_pred_right_eq_Ico`" (since := "2025-02-25")]
 theorem Icc_pred_right {b : ℕ} (h : 0 < b) : Icc a (b - 1) = Ico a b := by
   ext x
   rw [mem_Icc, mem_Ico, lt_iff_le_pred h]
 
+@[deprecated "Use `Finset.Ico_succ_succ_eq_Ioc`" (since := "2025-02-25")]
 theorem Ico_succ_succ : Ico a.succ b.succ = Ioc a b := by
   ext x
   rw [mem_Ico, mem_Ioc, succ_le_iff, Nat.lt_succ_iff]
 
+set_option linter.deprecated false in
 @[simp]
 theorem Ico_succ_singleton : Ico a (a + 1) = {a} := by rw [Ico_succ_right, Icc_self]
```

Comment:
Can't we just prove these using the new lemmas rather than adding a lint option here?

Reply:
Doing so increases imports quite a lot, which doesn't seem worth it?

Reply:
Once the now-deprecated lemmas are deleted, what's the plan for these ones then? Unless you're planning to generalise these too (which would probably be the best option), you'll either need to increase the imports at some point anyway.

Reply:
No, I'll need to change imports in downstream files, but those lemma can probably simply be deleted

Reply:
Deprecate it in favor of `Ico_succ_right_eq_Icc` + `Icc_self`, yes

Reply:
So you want `Ico_succ_right_eq_Icc` to be simp? Are you sure about this?

Reply:
I'm not sure about, no, which is why I have neither generalised nor deprecated the lemma in the current PR

Reply:
And that brings me back to the original question: What is the plan for these ones?

Reply:
"Think about them later"

Reply:
Let's leave a comment/todo so that whoever comes to remove the deprecated lemmas knows what to do with them.